fdbserver/ptxn/test/RealTLog.actor.cpp

```diff
@@ -249,41 +319,122 @@ ACTOR Future<Void> getTLogGroupActor(std::shared_ptr<TestDriverContext> pTestDri
 	pTLogDriverContext->tLogSet.tLogVersion = TLogVersion::V6;
 
 	pTLogDriverContext->dbInfo.logSystemConfig.tLogs.push_back(pTLogDriverContext->tLogSet);
-	pTLogDriverContext->ls = ILogSystem::fromServerDBInfo(pTLogDriverContext->logID, pTLogDriverContext->dbInfo, false);
+	PromiseStream<Future<Void>> promises;
+	pTLogDriverContext->ls =
+	    ILogSystem::fromServerDBInfo(pTLogDriverContext->logID, pTLogDriverContext->dbInfo, false, promises);
 
 	// start transactions
-	pTLogDriverContext->pTLogContext->TLogStarted.send(true);
-	Future<Void> commit = pTLogDriverContext->sendCommitMessages(pTestDriverContext);
-	Future<Void> peek = pTLogDriverContext->peekCommitMessages(pTestDriverContext);
+	pTLogDriverContext->pTLogContextList[0]->TLogStarted.send(true);
+	Future<Void> commit = pTLogDriverContext->sendCommitMessages();
+	Future<Void> peek = pTLogDriverContext->peekCommitMessages();
 	wait(commit && peek);
 
 	// tell tLog actor to initiate shutdown.
-	pTLogDriverContext->pTLogContext->realTLogTestCompleted.send(true);
+	pTLogDriverContext->pTLogContextList[0]->realTLogTestCompleted.send(true);
 
 	return Void();
 }
 
+// wait for all tLogs to be created. Then start actor to do push, then
+// start actors to do peeks, then signal transactions can start.
+ACTOR Future<Void> getProxyActor(std::shared_ptr<TLogDriverContext> pTLogDriverContext) {
+	// create tLog
+	state uint16_t processID = 0;
+	for (; processID < pTLogDriverContext->numLogServers; processID++) {
+		state std::shared_ptr<TLogContext> pTLogContext = pTLogDriverContext->pTLogContextList[processID];
+		bool isCreated = wait(pTLogContext->TLogCreated.getFuture());
+		ASSERT_EQ(isCreated, true);
+
+		// setup log system and tlog group
+		TLogSet tLogSet;
+		tLogSet.tLogs.push_back(OptionalInterface<TLogInterface>(pTLogContext->realTLogInterface));
+		tLogSet.tLogLocalities.push_back(LocalityData());
+		tLogSet.tLogPolicy = Reference<IReplicationPolicy>(new PolicyOne());
+		tLogSet.locality = 0;
+		tLogSet.isLocal = true;
+		tLogSet.tLogVersion = TLogVersion::V6;
+		pTLogDriverContext->dbInfo.logSystemConfig.tLogs.push_back(tLogSet);
+
+		// start transactions
+		pTLogContext->TLogStarted.send(true);
+	}
+
+	PromiseStream<Future<Void>> promises;
+	pTLogDriverContext->ls =
+	    ILogSystem::fromServerDBInfo(pTLogDriverContext->logID, pTLogDriverContext->dbInfo, false, promises);
+
+	state std::vector<Future<Void>> actors;
```

Comment:
Nit: this doesn't need to be a `state` variable, because it is not used after the `wait`.

Non-state variable can be destructed early, so it's slightly more memory efficient.